tools/topology/sof/pipe-amp-ref-capture.m4

```diff
@@ -0,0 +1,125 @@
+# Amplifier feedback reference Demux Pipeline
+#
+#  capture with demux for both echo reference to user space and the feedback to smart amplifier FW algorithm.
+#
+# Pipeline Endpoints for connection are :-
+#
+#	Capture Demux
+#	B1 (DAI buffer)
+#
+#
+#                          |
+#  host PCM_C <-- B0 <-- Demux(M) <-- B1 <-- DAI
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`buffer.m4')
+include(`pcm.m4')
+include(`pga.m4')
+include(`muxdemux.m4')
+include(`mixercontrol.m4')
+include(`bytecontrol.m4')
+
+ifdef(`SMART_TX_CHANNELS',`',`errprint(note: Need to define DAI TX channel number for sof-smart-amplifier
+)')
+ifdef(`SMART_RX_CHANNELS',`',`errprint(note: Need to define DAI RX channel number for sof-smart-amplifier
+)')
+ifdef(`SMART_FB_CHANNELS',`',`errprint(note: Need to define feedback channel number for sof-smart-amplifier
+)')
+
+ifdef(`SMART_PB_PPL_ID',`',`errprint(note: Need to define playback pipeline ID for sof-smart-amplifier
+)')
+ifdef(`SMART_PB_CH_NUM',`',`errprint(note: Need to define playback channel number for sof-smart-amplifier
+)')
+
+ifdef(`SMART_REF_PPL_ID',`',`errprint(note: Need to define Echo Ref pipeline ID for sof-smart-amplifier
+)')
+ifdef(`SMART_REF_CH_NUM',`',`errprint(note: Need to define Echo Ref channel number for sof-smart-amplifier
+)')
+
+dnl COMP_FORMAT_VALUE(FMT)
+     define(`COMP_FORMAT_VALUE',
+     `ifelse(
+         $1, `s16le', `0x00',
+         $1, `s24le', `0x01',
+         $1, `s32le', `0x02',
+         )')
+
+# note: number >= 16 not supported yet!
+define(`DEC2HEX',
+	`ifelse(
+		eval(`$1 < 10'), `1', `0x0'$1,
+		eval(`$1 < 16'), `1', concat(`0x0',eval($1, 16)),
+		)')
+
+
+CONTROLBYTES_PRIV(DEMUX_priv,
```

Comment:
Can you comment here how this was generated and what the config does.

Reply:
Sure.